modules/katello_activation_key.py

```diff
@@ -0,0 +1,248 @@
+                new_attr = getattr(new, field)
+                old_attr = getattr(old, field)
+                if old_attr is None or new_attr.id != old_attr.id:
+                    setattr(old, field, new_attr)
+                    needs_update = True
+            elif hasattr(old, field) and getattr(old, field) is not None and not hasattr(new, field):
+                setattr(old, field, None)
+                needs_update = True
+        return needs_update, old
+
+    def activation_key(self, name, organization, lifecycle_environment=None, content_view=None, subscriptions=[]):
+        updated = False
+        organization = self.find_organization(organization)
+
+        kwargs = {'name': name, 'organization': organization}
+
+        if lifecycle_environment:
+            kwargs['environment'] = self.find_lifecycle_environment(lifecycle_environment, organization)
+
+        if content_view:
+            kwargs['content_view'] = self.find_content_view(content_view, organization)
+
+        activation_key = self._entities.ActivationKey(self._server, **kwargs)
+        response = activation_key.search({'name', 'organization'})
+
+        if len(response) == 0:
+            activation_key = activation_key.create()
+            updated = True
+        elif len(response) == 1:
+            updated, activation_key = self.update_fields(activation_key, response[0], ['organization', 'environment', 'content_view'])
+            if updated:
+                activation_key.update()
+
+        if subscriptions is None:
+            subscriptions = []
+        desired_subscription_ids = map(lambda s: s.id, self.find_subscriptions(subscriptions, organization))
+        current_subscriptions = [entities.Subscription(self._server, **result)
+                                 for result in entities.Subscription(self._server).search_normalize(activation_key.subscriptions()['results'])]
+        current_subscription_ids = map(lambda s: s.id, current_subscriptions)
+
+        if set(desired_subscription_ids) != set(current_subscription_ids):
+            for subscription_id in set(desired_subscription_ids) - set(current_subscription_ids):
+                activation_key.add_subscriptions(data={'quantity': 1, 'subscription_id': subscription_id})
+            for subscription_id in set(current_subscription_ids) - set(desired_subscription_ids):
+                activation_key.remove_subscriptions(data={'subscription_id': subscription_id})
+            updated = True
+
+        return updated
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(required=False, type='bool', default=False),
+            name=dict(required=True, no_log=False),
+            organization=dict(required=False, no_log=False),
```

Comment:
shouldn't the org be required? otherwise it fails for me with:

```
FAILED! => {"changed": false, "failed": true, "msg": "No organization found for None"}
```